### ISSUE TITLE
Please add arm64 for iphonesimulator build in XCFramework

### ISSUE THREAD

**Issue by xiao99xiao** [voyagegroup/FluctSDK-iOS#6]
# TL;DR:

Please arm64 support for iphonesimulator build in your xcframework, like AdColony did https://github.com/AdColony/AdColony-iOS-SDK/tree/master/AdColony.xcframework , or every iOS project will have to abandon FluctSDK eventually.

# Summary

FluctSDK's lacking of arm64 support for iphonesimulator build makes any projects integrating FluctSDK and other libraries with cocoapods **unable** to be build and run for iPhone Simulator on M1 MacBook, only to receive a `Could not find module '<Other Library>' for target 'x86_64-apple-ios-simulator'; found: arm64, arm64-apple-ios-simulator` error from Xcode.

# Full Explanation

This is gonna be a long story, please read with patient.

## 1. Background

When there are only Intel CPUs on every Macbook, the iPhone simulators shipped with Xcode are all actually in i386 or x86_64 architecture, while all real iPhone devices are all among `arm64 arm64e armv7 armv7s` architecture.

So a xcframework, with its simulator-targeting framework built for only i386 and x86_64, and real-device-targeting framework built for only `arm64 arm64e armv7 armv7s` or even less, is very **appropriate**, at that time.

FluctSDK is still doing exact the same today, which is becoming a problem because of the coming of M1 MacBook.

## 2. M1 Macbook and arm64-apple-ios-simulator

Currently, Xcode on MacBook with Intel CPUs, runs iPhone simulators in x86_64 architecture, same like before, which is now called `x86_64-apple-ios-simulator`.

Mean while, Xcode on MacBook with Apple Silicon(Or M1), runs iPhone simulators defaultly in arm64 architecture, which is called `arm64-apple-ios-simulator`. This may be new to you but it is a obvious thing if you remember the Apple Silicon(M1) is an arm64 chip and the whole macOS on MacBook M1 is running on it.

Of course with Rosetta, we _can_ run Xcode and iPhone simulators on MacBook M1 in x86_64 mode. But it's always causing performance issue for everyone.

## 3. FluckSDK Issue

So, with the MacBook M1 and `arm64-apple-ios-simulator` here, what FluctSDK is still doing today is causing a big problem. To make it easy to reproduce, I'll describe the issue in a standard way as below:

### Environment

* MacBook with Apple Silicon
* macOS 12
* Xcode in native mode(not Rosetta)
* Latest Ruby installed by rbenv so it's built with full arm64 support.
* Cocoapods installed by brew in native mode(not executed with `arch x86_64` prefix shit)

### Steps to Reproduce

1. Create a new iOS project in Xcode
2. Install FluctSDK and any other library(Let's pick [Lottie](https://github.com/airbnb/lottie-ios) here) via Podfile
3. run `pod install --repo-update` in terminal.
4. Open the generated xcworkspace in Xcode.
5. Select build target to iPhone 13 Simulator.
6. Click Run button or hit CMD+R

### Expected Result

Project gets built successfully and runs in iPhone Simulator

### Actual Result/Error

Xcode reports Failure on compiling

`Could not find module 'Lottie' for target 'x86_64-apple-ios-simulator'; found: arm64, arm64-apple-ios-simulator`

https://github.com/airbnb/lottie-ios/issues/1293

## 4. But Xcode Said 'Lottie' in the Error……?

**Here comes the most hard part to explain:**

Though Xcode reported an error with Lottie blablabla, but the real cause for the compilation failure is on FluctSDK.

Consider if there is nothing wrong in the project, Xcode on Apple Silicon is supposed to build the project and its dependencies in arm64 arch, since its build target is `arm64-apple-ios-simulator`.

But what if one of its dependency, FluctSDK, [explictly claims it's not supporting arm64 on simulator](https://github.com/voyagegroup/FluctSDK-iOS/blob/6336f9df667397e92b7ada8aaa9799f36a3d9e90/FluctSDK.podspec#L18)?

The answer is, Xcode is going to be so mad that it have no choice but to try to compile for the only other simulator arch shipped inside: `x86_64-apple-ios-simulator`

**Why would this be wrong?**

Because for the DEBUG scheme of any projects, the `Build Active Architecture Only` is always `YES`, and the only active architecture during our compilation, is `arm64-apple-ios-simulator` of course, which means other archs including `x86_64-apple-ios-simulator` will not be built at all. So no wonder Xcode complains `Could not find module 'Lottie' for target 'x86_64-apple-ios-simulator'`

## 5. Who is Responsible for this?

You might say, "Why don't you just set the `Build Active Architecture Only` to `NO`?" . My answer is, compiling for other inactive archs is wasting everyone's life as it drastically extends the compilation time to 3 or 4 times. What's more important, why waste everyone's time while you can easily fix it? Yeah I mean you, FluctSDK.

## 6. What if You Don't Fix it?

Admit it or not, people are migrating to Apple Silicon rapidly. Soon, every new iOS project will be compiled on a Apple Silicon mac.

For now, many people are confused by Xcode's strange error description, unable to find out the real reason behind is frameworks like FluctSDK that do not support arm64 on simulator. As if you [search `Could not find module for target 'x86_64-apple-ios-simulator'; found: arm64, arm64-apple-ios-simulator` on github](https://github.com/search?o=desc&p=1&q=Could+not+find+module+for+target+%27x86_64-apple-ios-simulator%27&s=&type=Issues) you'll find numerous issues posted on varies library's issue page. But sooner or later people will figure it out.

At that time, every iOS project, installed with FluctSDK, will not be able to debug on iPhone Simulator without breaking several bones of its developer. And eventually we would give up installing FluctSDK and use other Ad frameworks like AdColony that already provides universal framework support.

## 7. How to Fix it?

- Create a arm64 version of framework targeting iPhone Simulator and add it to xcframework;
- Remove the `pod_target_xcconfig` and `user_target_xcconfig` in podspec file.

I'm not an expert on this but if you need some reference then may be [this](https://developer.apple.com/forums/thread/666335?answerId=662473022#662473022) could help you.

I would be very happy to help if any further info is needed.

**Comment by arimura** [comment]
@xiao99xiao 
Sorry for the late reply. FluctSDK supports arm64 iPhone simulator in below version.
https://voyagegroup.github.io/FluctSDK-Doc/#/fluctsdk-ios/others/release-note?id=v6250-20230331

**Comment by xiao99xiao** [comment]
@arimura 
Congratulations. 
Though I'm already relieved from the iOS app dev stuff, your update will make many developers' life a whole lot easier.
